pylint/checkers/typecheck.py

```diff
@@ -1720,9 +1720,18 @@ def visit_subscript(self, node):
             return
 
         inferred = safe_infer(node.value)
+
         if inferred is None or inferred is astroid.Uninferable:
             return
 
+        if inferred.decorators:
+            first_decorator = helpers.safe_infer(inferred.decorators.nodes[0])
+            if isinstance(first_decorator, astroid.ClassDef):
+                inferred = first_decorator.instantiate_class()
+            else:
+                return  # It would be better to handle function
+                # decorators, but let's start slow.
+
```

Comment:
The following lines take into account my remarks. Fill free to use it as it is or change everything. It is just a suggestion.
```suggestion
        if supported_protocol(inferred):
            return

        if inferred.decorators:
            # It there are decorators, maybe one of them enable the support of the protocol
            for deco_node in inferred.decorators.nodes:
                decorator = helpers.safe_infer(deco_node)
                if (isinstance(decorator, astroid.ClassDef) and 
                    supported_protocol(decorator.instantiate_class())):
                    return

        # The protocol is not supported neither by the node nor through the decorators 
        self.add_message(msg, args=node.value.as_string(), node=node.value)
```